### PR TITLE
IGNITE-11199 Add extra logging for client-server connections in TCP discovery.

### PR DESCRIPTION
**Pull request by 6uest**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java [hunk 1]
```diff
@@ -433,9 +433,30 @@
...
                 else {
-                    if (log.isInfoEnabled())
+                    if (log.isInfoEnabled()) {
+                        String serverNodeId = "";
+                        String consistentId = "";
+
+                        try {
+                            Collection<ClusterNode> remoteNodes = getSpiContext().remoteNodes();
+
+                            for (ClusterNode node : remoteNodes) {
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java]: Please merge two lines in single. i.e. `for (ClusterNode node : getSpiContext().remoteNodes())`

File: modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java [hunk 2]
```diff
@@ -433,9 +433,30 @@
...
+                                if (U.toInetAddresses(node).contains(ses.remoteAddress().getAddress()) && !node.isClient()) {
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java]: We could get not all possible remote nodes, if we will have a few server nodes on one host.

File: modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java [hunk 3]
```diff
@@ -433,9 +433,30 @@
...
+                                    consistentId = node.consistentId().toString();
+
+                                    break;
+                                }
+                            }
+                        } catch (IgniteCheckedException e) {
+                            log.warning("Unable to acquire remote node ID from SpiContext: " + e, e);
+                        }
+
                         log.info("Established outgoing communication connection [locAddr=" + ses.localAddress() +
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java]: Please wrap `log.info()` to `if(log.isInfoEnabled())` condition.

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 4]
```diff
@@ -6014,7 +6014,8 @@ public void onInterruption() {
 
             if (log.isInfoEnabled())
                 log.info("Started serving remote node connection [rmtAddr=" + rmtAddr +
-                    ", rmtPort=" + sock.getPort() + ']');
+                    ", localNodeId=" + locNodeId +
```
Comment (antonovsergey93) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java]: We could try to detect which node is connecting to this node and print info about remote node.

File: modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java (hunk 3 above)
Comment (6uest) [reply modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java]: But `log.info()` already wrapped

Comment (antonovsergey93) [reply modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java]: Ok. I look over it.

**Pull request closed without merging**